providers/dns/script/script.go

```diff
@@ -0,0 +1,75 @@
+// Package script implements a manual DNS provider which runs a script for
+// adding/removing the DNS record.
+//
+// The file name of the external script is specified in the environment
+// variable SCRIPT_PATH. When it is run by lego, three command-line parameters
+// are passed to it: The action ("add" or "remove"), the fully-qualified domain
+// name, the value for the record and the TTL.
+//
+// For example, requesting a certificate for the domain 'foo.example.com' can
+// be achieved by calling lego as follows:
+//
+//  SCRIPT_PATH=./update-dns.sh \
+//    lego --dns script \
+//    --domains foo.example.com \
+//    --email invalid@example.com run
+//
+// It will then call the script './update-dns.sh' with like this:
+//
+//  ./update-dns.sh "add" "_acme-challenge.foo.example.com." "MsijOYZxqyjGnFGwhjrhfg-Xgbl5r68WPda0J9EgqqI" "120"
+//
+// The script then needs to make sure the record is inserted. When it returns
+// an error via a non-zero exit code, lego aborts.
+//
+// When the record is to be removed again, the script is called with the first
+// command-line parameter set to "remove" instead of "add".
+package script
```

Comment:
Technically this can be any executable file or program, not just a script. Is there a better name we can use?

Reply:
I thought about this as well, but liked the other names even less:
 * `manual`
 * `ext`
 * `external`
 * `program`

The name `script` is the least ugly I came up with.

Reply:
It's not a major thing, just want to get the optimal name the first time. How about something more general like "delegated" or "exec" (maybe not exec for a package name, of course).

If those aren't great then maybe we can just keep it the same, but I kinda like "delegated" or even "external".

Reply:
Hm, then I'd prefer `exec`. The package can be called anything, we're just talking about the user interface, right?